File: server/CLup.Businesses/Business.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

using CLup.Context;
using CLup.Employees;
using CLup.TimeSlots;

namespace CLup.Businesses
{
    public class Business : BaseEntity
    {
        public string Id { get; set; }

        [Required]
        [StringLength(50)]
        public string Name { get; set; }

        [Required]
        public string OwnerEmail { get; set; }

        [Required]
        public string Zip { get; set; }

        [Required]
        public string Address { get; set; }

        [Required]
        public double Longitude { get; set; }

        [Required]
        public double Latitude { get; set; }

        [Required]
        public BusinessType Type { get; set; }

        [Required]
        public int Capacity { get; set; }

        [Required]
        public string Opens { get; set; }

        [Required]
        public string Closes { get; set; }

        [Required]
        public int TimeSlotLength { get; set; }

        public IEnumerable<TimeSlot> TimeSlots { get; set; }

        public IEnumerable<Employee> Employees { get; set; }

    }
}

File: src/CLup.WebUI/Client/src/features/business/Business.ts
import {Address} from '../../shared/services/AddressService';
import {DTO, HasAddress} from '../../shared/models/General';
import { BookingDTO } from '../booking/Booking';
import { EmployeeDTO } from './employee/Employee';
import { MessageDTO } from '../message/Message';
import { TimeSlotDTO } from './timeslot/TimeSlot';


export interface BusinessDTO extends DTO, HasAddress {
    name: string;
    type: string;
    timeSlotLength: number | string;
    capacity: number | string;
    opens: string;
    closes: string;
    businessHours?: string;
    ownerEmail?: string;
    bookings?: BookingDTO[]
    employees?: EmployeeDTO[]
    messages?: MessageDTO[]
    timeSlots?: TimeSlotDTO[]
}

export type BusinessDataDTO = {
    numberOfBookings: number;
    numberOfTimeSlots: number;
    numberOfEmployees?: number;
};

export function updateBusiness(
    updatedBusiness: BusinessDTO,
    business: BusinessDTO,
    address: Address
) {
    updatedBusiness.longitude = address?.longitude ?? business.longitude;
    updatedBusiness.latitude = address?.latitude ?? business.latitude;
    updatedBusiness.city = address?.city ?? business.city;
    updatedBusiness.zip = address?.zip ?? business.zip;

    updatedBusiness.opens = updatedBusiness.opens.replace(':', '.');
    updatedBusiness.closes = updatedBusiness.closes.replace(':', '.');

    return updatedBusiness;
}
